client/pages/admin/signin.js

```diff
@@ -27,15 +27,24 @@ const AdminSignin = () => {
         identifier: data.identifier,
         password: data.password,
       }),
-    });
-    const responseData = await response.json();
-    setToken(responseData, router);
-
-    const userCookie = Cookies.get('username');
-    if (userCookie) {
-      setUser(userCookie);
-      router.push('/admin/meetups');
-    }
+    })
+      .then((response) => {
+        return response.json()
+      })
+      .then((jsonResponse) => {
```

Comment:
Could you use `catch` instead?

https://medium.com/@lucymarmitchell/using-then-catch-finally-to-handle-errors-in-javascript-promises-6de92bce3afc